app/src/main/java/cm/aptoide/pt/wallet/WalletInstallAnalytics.kt

```diff
@@ -39,9 +52,37 @@ class WalletInstallAnalytics(val downloadAnalytics: DownloadAnalytics,
     setupDownloadAnalyticsEvents(download, campaignId, abTestGroup, downloadAction,
         AnalyticsManager.Action.CLICK, offerResponseStatus)
     installAnalytics.installStarted(download.packageName, download.versionCode,
-        AnalyticsManager.Action.INSTALL, AppContext.APPVIEW,
+        AnalyticsManager.Action.INSTALL, AppContext.WALLET_INSTALL_ACTIVITY,
         downloadStateParser.getOrigin(download.action), campaignId, abTestGroup,
-        downloadAction != null && downloadAction == DownloadModel.Action.MIGRATE, download.hasAppc())
+        downloadAction != null && downloadAction == DownloadModel.Action.MIGRATE,
+        download.hasAppc())
+
+  }
+
+  fun sendClickOnInstallButtonEvent(packageName: String, applicationPublisher: String) {
+    val map = HashMap<String, Any>()
+    map[TYPE] = "Install"
+    map[APPLICATION_NAME] = packageName
+    map[APPLICATION_PUBLISHER] = applicationPublisher
+    analyticsManager.logEvent(map, CLICK_INSTALL, AnalyticsManager.Action.CLICK,
+        VIEW_CONTEXT)
+  }
+
+  fun setupHistoryTracker() {
```

Comment:
Why is this here? If we want to register the screen, shouldn't it be in the activity?

Reply:
This was done by suggestion from João.
Ideally we would have a base activity that would do this management, just like we have the NavigationTrackerFragment, but João said it was not worth the effort, considering we only have one more activity which is the MainActivity. That is why every other navigation track management is done in the respective Fragments.
For this case we did it here in the wallet analytics because we needed to add a flag (shouldRegister) and we had to add the dependency of NavigationTracker. It didn't make sense to make the activity dirty because of this.

Reply:
👍